pkg/pipelines/git/repository.go

```diff
@@ -0,0 +1,134 @@
+package git
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/jenkins-x/go-scm/scm"
+	"github.com/jenkins-x/go-scm/scm/factory"
+)
+
+// Repository represent a Git repository ofa specific Git repository URL
+type Repository struct {
+	*scm.Client
+
+	// name is the repository name of the form <user>/<repository>
+	name string
+}
+
+// NewRepository creates a new Git reposiory object
+func NewRepository(rawURL, token string) (*Repository, error) {
+
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	driverName, err := getDriverName(parsedURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := factory.NewClient(driverName, "", token)
+	if err != nil {
+		return nil, err
+	}
+
+	repoName, err := getRepoName(parsedURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Repository{name: repoName, Client: client}, nil
+}
+
+// ListWebhooks returns a list of webhook IDs of the given listener in this repository
+func (r *Repository) ListWebhooks(listenerURL string) ([]string, error) {
+
+	hooks, _, err := r.Client.Repositories.ListHooks(context.Background(), r.name, scm.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	ids := []string{}
+	for _, hook := range hooks {
+		if hook.Target == listenerURL {
+			ids = append(ids, hook.ID)
+		}
+	}
+
+	return ids, nil
+}
+
+// DeleteWebhooks deletes all webhooks that associate with the given listener in this repository
+func (r *Repository) DeleteWebhooks(ids []string) ([]string, error) {
```

Comment:
This comment doesn't sound quite right, it seems to be deleting all webhooks with the IDs provided?